### PR TITLE
Add example for SSL client auth.

### PR DESCRIPTION
**Pull request by danielgtaylor**
## What do these changes do?

For my use-case I needed to set up the SSL context to do client auth. It will not work unless I pass in the right purpose when creating the context. I searched around a long time and couldn't find anything about it, so I'm adding it to the docs to help others (and my own forgetful future self).

## Are there changes in behavior for the user?

No

## Related issue number

None

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: docs/client_advanced.rst [hunk 1]
```diff
@@ -424,7 +424,15 @@ same thing as the previous example, but add another call to
                              '/path/to/client/private/device.key')
   r = await session.get('https://example.com', ssl=sslcontext)
 
-There is explicit errors when ssl verification fails
+If you need to use client-side SSL certificates for authentication
+you can create the :class:`ssl.SSLContext` with that purpose::
+
+  sslcontext = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
```
Comment (asvetlov) [inline docs/client_advanced.rst:430]: Should it be `ssl.Purpose.SERVER_AUTH`?
https://docs.python.org/3/library/ssl.html#ssl.Purpose.SERVER_AUTH

Comment (danielgtaylor) [reply docs/client_advanced.rst:430]: Here's a good article that describes my use case:

https://medium.com/@sevcsik/authentication-using-https-client-certificates-3c9d270e8326

TL;DR: The server authenticates the client during the SSL handshake by using the client's cert. This is different from the "normal" flow where the client checks the server's cert. In fact, in a development environment we have required client certs that are used to talk to servers with self-signed certs so the *only* check is that the client sends a valid cert that the server is preconfigured to accept.

Comment (asvetlov) [reply docs/client_advanced.rst:430]: Got you. The problem is: you are trying to describe a working and useful but not common use case.
I afraid added 10 lines produce more confusion than help.
A much more comprehensive text is needed to describe a motivation for the mode, like the mentioned article.

But I have a feeling that aiohttp doc is not a good place for such articles, the library *uses* `ssl`, describes very trivial cases but doesn't dig into SSL too deep -- the subject is out of `aiohttp` scope (like basic Python programming and many other things).

**Comment by danielgtaylor** [comment]
Closing this based on feedback that this use case is too specific and uncommon to warrant going into the docs. Thanks for considering the change!

**Comment by asvetlov** [comment]
Welcome and thanks for the PR!
Even not merged pull requests are valuable, I've learned a new certificates game :)

**Comment by webknjaz** [comment]
I actually think some info would be useful. While it may be uncommon for generic apps, I believe it's very useful in corporate envs, for example.

**Comment by asvetlov** [comment]
A FAQ section?

**Comment by webknjaz** [comment]
Yep, at least there. Let's raise people's awareness :)
